roles/redhat_unsubscribe/tasks/main.yml

```diff
@@ -0,0 +1,42 @@
+---
+# vim: set ft=ansible:
+#
+# NOTE: This role should only be used on RHEL systems
+#
+# We get the neccessary info to insert the 'unconfigured-state' field into the origin
+# file only when we need to.  Otherwise, we just use the sub-man command to remove
+# all the subscriptions.
+#
+- block:
+  - name: Get the current commit to find the correct origin file to manipulate later...
+    command: rpm-ostree status --json
+    register: rpm_output
+
+  - name: Convert deployment information into jinja2 json
+    set_fact:
+      json: "{{ rpm_output.stdout|from_json }}"
+
+  - name: Set version and commit if deployment 0 is booted
+    set_fact:
+      booted_commit: "{{ json['deployments'][0]['checksum'] }}"
+    when: json['deployments'][0] is defined and json['deployments'][0]['booted']
+
+  - name: Set version and commit if deployment 1 is booted
+    set_fact:
+      booted_commit: "{{ json['deployments'][1]['checksum'] }}"
+    when: json['deployments'][1] is defined and json['deployments'][1]['booted']
+  when:
+    - unconfigured_state is defined
+    - unconfigured_state
+
+- name: Remove all subscriptions
+  command: subscription-manager remove --all
```

Comment:
Dang.  I actually recall hitting this yesterday and thinking I should fix it.  Will address it shortly.